src/utils.ts

```diff
@@ -67,54 +83,40 @@ function stringifyWebhooksToBase64(webhooks) {
     return btoa(String.fromCharCode(...uint8Array));
 }
 
-let gzipPromise;
+let gzipPromise: ReturnType<typeof util['promisify']>;
 if (isNode()) gzipPromise = util.promisify(zlib.gzip);
+
 /**
  * Gzip provided value, otherwise returns undefined.
- * @param value
- * @return {Promise<undefined|Buffer>}
  */
-async function maybeGzipValue(value) {
+export async function maybeGzipValue(value: unknown): Promise<Buffer | undefined> {
     if (!isNode()) return;
     if (value == null) return;
 
     // Request compression is not that important so let's
     // skip it instead of throwing for unsupported types.
     const areDataStringOrBuffer = (typeof value === 'string') || Buffer.isBuffer(value);
-    const areDataLargeEnough = Buffer.byteLength(value) >= MIN_GZIP_BYTES;
+    const areDataLargeEnough = Buffer.byteLength(value as string) >= MIN_GZIP_BYTES;
```

Comment:
Shouldn't this be

```ts
if (!isNode()) return;
if (typeof value !== 'string' || !Buffer.isBuffer(value)) return;

const areDataLargeEnough = Buffer.byteLength(value) >= MIN_GZIP_BYTES;
if (areDataLargeEnough) {
	return gzipPromise(value);
}
```

This way it won't throw on `maybeGzipValue({})`

Reply:
You're right, that would be better

Reply:
The `if (value == null) return;` check seems to be unnecessary now